### PR TITLE
Featurize transcription modes

### PR DESCRIPTION
**Pull request by jcdyer**
## Proposed changes

Add feature flags for `live` and `prerecorded` ASR.  This helps non-streaming users not have to import tungstenite.  I'm open to not doing things this way.  I'm thinking of also following up with a little more specificity about which tokio features to include for similar reasons.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) doc
- [ ] I have added tests and/or examples that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -34,3 +34,8 @@ proc-macro2 = "1.0.60"
 pkg-config = "0.3.27"
 cpal = "0.13"
 crossbeam = "0.8"
+
+[features]
+default = ["prerecorded", "live"]
+live = ["dep:tungstenite", "dep:tokio-tungstenite"]
```
Comment (andrewhalle) [inline Cargo.toml:40]: Use of `dep:` bumps our MSRV to 1.60. I don't think we document an MSRV, but we should and it'll need to be at least 1.60.

**Review by andrewhalle — APPROVED** [review]
I swear we had these feature flags already, they must have been removed at some point in the past

File: Cargo.toml (hunk 1 above)
Comment (jcdyer) [reply Cargo.toml:40]: Whoa.  rust 1.60.0 is from April 2022.

Comment (jcdyer) [reply Cargo.toml:40]: I see the feature flags in a PR that never got merged.  Maybe that's what you were thinking of.

Comment (andrewhalle) [reply Cargo.toml:40]: That must be it

**Comment by jcdyer** [comment]
Oops.  This PR breaks the examples.  (Found doing `cargo test --no-default-features`.  I guess I'll need to annotate the required features for each one.

**Comment by jcdyer** [comment]
Also updated CI to run `check` with no features and with each feature in isolation.

**Review by andrewhalle — APPROVED** [review]

**Pull request merged**